### ISSUE TITLE
warning: cannot find deviceFun for cublas function. 

### ISSUE THREAD

**Issue by hanloveland** [gpgpu-sim/gpgpu-sim_distribution#204]
Hello I am using gpgpu-sim to run a gemm only using cublas. The gpgpusim config is SM7-TITANV. 

When I run a simple gemm program, I get a segmentation fault, as shown below. When checking the log, segmentation occurs while the `cublassgemm` function calls `sgemm_ 32x32x32_NN_vec`. 

![image-20201021225507077](https://user-images.githubusercontent.com/58060090/96733129-14645600-13f4-11eb-927b-7d591cf4d8a2.png)
Also, there is a warning message in the log that the devicefun of ``sgemm_32x32x32_NN_vec` could not be found. 

![image-20201021230231611](https://user-images.githubusercontent.com/58060090/96733200-2645f900-13f4-11eb-998f-1d3986ce4876.png)

Probably, this error seems to occur because there is no ptx of `sgemm_32x32x32_NN_vec`. However, in some matrix dimensions (e.g., M: 512, N:512, K: 1), the cublas seem to operate normally by calling a function such as `**Z13gemmk1_kernelIfLi256ELi5ELb0ELb0ELb0ELb0EEv18cublasGemmk1ParamsIT_EPKS1_S4_PS1**` as below.

![image-20201021230231611](https://user-images.githubusercontent.com/58060090/96733374-51304d00-13f4-11eb-980b-ad000dfeb333.png)

Anyone know the reason for this error and how to fix it? Could anyone help me out with this issue? Thanks a lot in advance!

**Comment by mattsinc** [comment]
There have been several prior issues posted on this.  But, broadly, it relates to what version of CUDA you are using, and what version of GPGPU-Sim you are using.

NVIDIA stopped embedding the PTX in their libraries after CUDA 8.  So, if you are using a newer version of CUDA, you will see failures like this because, once the program reaches a kernel it doesn't have PTX for, GPGPU-Sim doesn't know what to do and fails.  If you want to run newer versions of CUDA with cuBLAS/cuDNN/etc., you'll need to use the AccelSim framework.

If are using CUDA 8 (or earlier), you need to statically link the libraries (e.g., cublas_static) into your program.  This will ensure that the PTX GPGPU-Sim needs is included in the binary (although it will make the binary very large).  When using CUDA 8, if you see this failure, it means you didn't link the binary in properly.

Hope this helps,
Matt
